x-pack/plugins/observability_solution/inventory/public/components/search_bar/index.tsx

```diff
@@ -51,11 +54,41 @@ export function SearchBar() {
     syncSearchBarWithUrl();
   }, [syncSearchBarWithUrl]);
 
+  const registerSearchSubmittedEvent = useCallback(
+    ({
+      searchQuery,
+      searchIsUpdate,
+      searchEntityTypes,
+    }: {
+      searchQuery?: Query;
+      searchEntityTypes?: string[];
+      searchIsUpdate?: boolean;
+    }) => {
+      telemetry.reportEntityInventorySearchQuerySubmitted({
+        kuery_fields: getKqlFieldsWithFallback(searchQuery?.query as string),
```

Comment:
Nit: Can we do 
```suggestion
        kuery_fields: getKqlFieldsWithFallback(searchQuery?.query ?? ''),
```
instead? To be safe that we won't pass `undefined` there.

Reply:
yeah, but I guess this has to be done in another PR and backported. Changing it here will lead to differences between versions.